### PR TITLE
Modernizing plugin structure

### PR DESCRIPTION
**Pull request by crenan**
- Use parent POM 4.53 instead of 2.32.
- Require Jenkins 2.361.4 or later. Simplify testing for plugin maintainers.
- Move plugin description to index.jelly.
- Deeper spotbugs checks. Increase chance of detecting bugs.
- Update SCM URL.
- Automate dependency updates.
- Publish incremental development artifacts.
- Added release drafter to simplify changelog creation.
- Update version of dependencies.
- Update plugin README.md.

Followed the [Modernizing Jenkins Plugins](https://www.youtube.com/watch?v=Fev8KfFsPZE) video series.

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -3,15 +3,14 @@
     <parent>
         <groupId>org.jenkins-ci.plugins</groupId>
         <artifactId>plugin</artifactId>
-        <version>4.55</version>
+        <version>4.53</version>
```
Comment (slide) [inline pom.xml:6]: Why downgrading here?

File: pom.xml [hunk 2]
```diff
@@ -49,14 +48,12 @@
...
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>structs</artifactId>
-            <version>1.2</version>
+            <version>324.va_f5d6774f3a_d</version>
         </dependency>
         <dependency>
             <groupId>org.bouncycastle</groupId>
             <artifactId>bcprov-jdk15on</artifactId>
-            <version>1.47</version>
-            <scope>test</scope>
```
Comment (slide) [inline pom.xml]: Why removing the test scope here?

File: pom.xml [hunk 3]
```diff
@@ -49,14 +48,12 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>structs</artifactId>
-            <version>1.2</version>
+            <version>324.va_f5d6774f3a_d</version>
```
Comment (slide) [inline pom.xml:51]: Might be good to introduce the bom so that a specific version doesn't need to be here.

**Review by slide — CHANGES_REQUESTED** [review]
Just a couple of questions.

File: pom.xml (hunk 1 above)
Comment (slide) [reply pom.xml:6]: Sorry, was looking at an intermediate change or something, it showed downgrading from 4.55 to 4.53

File: pom.xml (hunk 3 above)
Comment (crenan) [reply pom.xml]: Added BOM.

File: pom.xml [hunk 4]
```diff
@@ -26,8 +26,9 @@
 
     <developers>
         <developer>
-            <id>slide_o_mix</id>
+            <id>slide</id>
```
Comment (slide) [inline pom.xml:29]: I think this is supposed to match the Jenkins infra username, not the GitHub username. So, in this case, my Jenkins username is slide_o_mix.

**Comment by slide** [comment]
I need to merge #21 before this will build correctly

**Comment by slide** [comment]
I created a shadow PR to verify this since the Jenkinsfile updates won't be picked up as your account is not a trusted account. See #22

**Comment by slide** [comment]
Thanks for the updates!

**Pull request merged**